pkg/debug/spec.sh

```diff
@@ -1,11 +1,73 @@
 #!/bin/sh
+# This script creates an initial hardware model file
+# The output is conservative when it comes to assignment groups in that
+# it does not form any assignment groups but merely checks whether there are
+# multiple functions on the same controller, and in that case the assignment
+# group is set to empty.
 
 if [ "$(uname -m)" = x86_64 ]; then
    ARCH=2
 else
    ARCH=4
 fi
 
+# log_to_dev returns the bus:controller without the function value
+# $1 is the PciLong value
+long_to_dev() {
+    #shellcheck disable=SC2039
```

Comment:
I highly recommend moving these kinds of #shellcheck disable right under `#!/bin/sh` so that it works globaly and you don't have to repeat them (btw https://www.shellcheck.net/ works super well for interactive experiments)

Reply:
Done